File: Taller_3V2/app/src/main/java/com/example/taller_3v2/MainActivity.java
package com.example.taller_3v2;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import com.example.taller_3v2.databinding.ActivityMainBinding;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;

public class MainActivity extends AppCompatActivity {

    ActivityMainBinding binding;

    private FirebaseAuth mAuth;
    FirebaseUser user = FirebaseAuth.getInstance().getCurrentUser();
    FirebaseDatabase database = FirebaseDatabase.getInstance();
    DatabaseReference myRef;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        binding.loginMBTN.setVisibility(View.VISIBLE);
        binding.RegisterBTN.setVisibility(View.VISIBLE);

        binding.loginMBTN.setOnClickListener(view -> {
            startActivity(new Intent(getApplicationContext(), LoginActivity.class));
            finish();
        });
        binding.RegisterBTN.setOnClickListener(view -> {
            startActivity(new Intent(getApplicationContext(), RegistroActivity.class));
            finish();
        });
    }
}

File: Quiz_RestCountries/app/src/main/java/com/example/quiz_restcountries/MainActivity.kt
package com.example.quiz_restcountries

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.example.quiz_restcountries.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private lateinit var binding : ActivityMainBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding =ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)


        binding.EuroEcon.setOnClickListener {
            startActivity(Intent(baseContext,EuroZoneActivity::class.java))
        }
        binding.YenCurrency.setOnClickListener {
            startActivity(Intent(baseContext,ActivityYenCapitals::class.java))
        }
        binding.FrenchCap.setOnClickListener{
            startActivity(Intent(baseContext,FrenchCapitalsActivity::class.java))
        }
        binding.EconZone.setOnClickListener {
            startActivity(Intent(baseContext,EconZoneActivity::class.java).apply {
                putExtra("Zona", binding.spinner.selectedItem.toString())
            })
        }
    }
}